google/cloud/storagecontrol/v2/samples/storage_control_folder_samples.cc

```diff
@@ -0,0 +1,219 @@
+    // Start a rename operation and block until it completes. Real applications
+    // may want to setup a callback, wait on a coroutine, or poll until it
+    // completes.
+    auto renamed = client.RenameFolder(name, dest_folder_id).get();
+    if (!renamed) throw std::move(renamed).status();
+
+    std::cout << "Renamed: " << source_folder_id << " to: " << dest_folder_id
+              << "\n";
+  }
+  // [END storage_control_rename_folder]
+  (std::move(client), argv.at(0), argv.at(1), argv.at(2));
+}
+
+void AutoRun(std::vector<std::string> const& argv) {
+  namespace examples = google::cloud::testing_util;
+  namespace storagecontrol = google::cloud::storagecontrol_v2;
+  if (!argv.empty()) throw examples::Usage{"auto"};
+  examples::CheckEnvironmentVariablesAreSet(
+      {"GOOGLE_CLOUD_CPP_STORAGE_TEST_FOLDER_BUCKET_NAME"});
+  auto const bucket_name =
+      google::cloud::internal::GetEnv(
+          "GOOGLE_CLOUD_CPP_STORAGE_TEST_FOLDER_BUCKET_NAME")
+          .value();
+
+  auto client = storagecontrol::StorageControlClient(
+      storagecontrol::MakeStorageControlConnection());
+  auto generator = google::cloud::internal::DefaultPRNG(std::random_device{}());
+  auto const prefix = "storage-control-samples";
+  auto const folder_id = prefix + std::string{"-"} +
+                         google::cloud::internal::Sample(
+                             generator, 32, "abcdefghijklmnopqrstuvwxyz");
+  auto const dest_folder_id = prefix + std::string{"-"} +
+                              google::cloud::internal::Sample(
+                                  generator, 32, "abcdefghijklmnopqrstuvwxyz");
+  auto const create_time_limit =
+      std::chrono::system_clock::now() - std::chrono::hours(48);
+  // This is the only example that cleans up stale folders. The examples run in
+  // parallel (within a build and across the builds), having multiple examples
+  // doing the same cleanup is probably more trouble than it is worth.
+  std::cout << "\nRemoving stale folders for examples" << std::endl;
+  RemoveStaleFolders(client, bucket_name, prefix, create_time_limit);
+
+  std::cout << "\nRunning CreateFolder() example" << std::endl;
+  CreateFolder(client, {bucket_name, folder_id});
+
+  std::cout << "\nRunning GetFolder() example" << std::endl;
+  GetFolder(client, {bucket_name, folder_id});
+
+  std::cout << "\nRunning ListFolders() example" << std::endl;
+  ListFolders(client, {bucket_name});
+
+  std::cout << "\nRunning RenameFolder() example" << std::endl;
+  RenameFolder(client, {bucket_name, folder_id, dest_folder_id});
+
+  std::cout << "\nRunning DeleteFolder() example" << std::endl;
+  DeleteFolder(client, {bucket_name, dest_folder_id});
+}
+
+}  // namespace
+
+int main(int argc, char* argv[]) {
+  using google::cloud::testing_util::Example;
+  namespace storagecontrol = google::cloud::storagecontrol_v2;
+  using ClientCommand =
+      std::function<void(google::cloud::storagecontrol_v2::StorageControlClient,
```

Comment:
nit: just say `storagecontrol::StorageControlClient` since you already defined the namespace alias.